core/src/main/java/com/rudderstack/android/sdk/core/RudderDataResidency.java

```diff
@@ -0,0 +1,130 @@
+     *
+     * <p>Preference of dataPlane URL is decided as follows:</p>
+     * <ul>
+     *     <li>First look for the Residency server (which was set while SDK initialisation) in the source config.</li>
+     *     <li>If Residency server was not set while SDK initialisation then fallback to default US residency server.</li>
+     *     <li>If residency server list is empty or null then fallback to dataPlane URL (which was set while SDK initialisation).</li>
+     *     <li>If dataPlane URL was not passed while SDK initialisation then as a last resort fallback to the default dataPlane URL.</li>
+     * </ul>
+     * <p>
+     * Default Residency server is US.
+     * <p>
+     * Use the below code snippet to set the Data Residency:
+     * <pre>
+     *     {@code
+     *     rudderClient = RudderClient.getInstance(
+     *             this,
+     *             WRITE_KEY,
+     *             RudderConfig.Builder()
+     *                 .withResidencyServer(ResidencyServer.US)
+     *                 .build()
+     *     )
+     *      }
+     * </pre>
+     */
+    void handleDataPlaneUrl() {
+        if (residencyServer == ResidencyServer.US) {
+            handleDefaultServer();
+        } else {
+            handleOtherServer(residencyServer);
+        }
+    }
+
+    /**
+     * Handle all server other than US.
+     * If passed server is not present then fallback to US server.
+     *
+     * @param residencyServer Residency server set while SDK initialisation
+     */
+    @VisibleForTesting
+    void handleOtherServer(@NonNull ResidencyServer residencyServer) {
+        // TODO: Decide upper or lower case
+        String dataResidencyUrl = getDataResidencyUrl(residencyServer.name());
+        if (dataResidencyUrl != null) {
+            config.setDataPlaneUrl(dataResidencyUrl);
+        } else {
+            handleDefaultServer();
+        }
+    }
+
+    /**
+     * Handle default US server
+     *
+     * If default US server is also not present then no need to do any extra things,
+     * as dataPlaneUrl logic is already handled while config object is build.
+     */
+    @VisibleForTesting
+    void handleDefaultServer() {
+        // TODO: Decide upper or lower case
+        String dataResidencyUrl = getDataResidencyUrl(ResidencyServer.US.name());
+        if (dataResidencyUrl != null) {
+            config.setDataPlaneUrl(dataResidencyUrl);
+        }
+    }
+
+    /**
+     * Look for the region in the data residency list.
+     *
+     * @param region Data Residency region set while SDK initialisation.
+     * @return Corresponding residency URL if present else null.
+     */
+    @VisibleForTesting
+    @Nullable
+    String getDataResidencyUrl(@Nullable String region) {
+        if (region != null)
+            if (isNotEmpty(dataResidencyUrls)) {
+                for (String key : dataResidencyUrls.keySet()) {
```

Comment:
We can get rid of this loop here, once we had decided on the casing from the control plane end

Reply:
Right, will update accordingly. Once we've clear picture on the casing.
For now I'll not change anything.

Reply:
@1abhishekpandey , just heard from @pallabmaiti , that let's check for lowercase only for now

Reply:
Sure, will implement that.